src/main/kotlin/org/rust/ide/actions/RustGoToSuperModAction.kt

```diff
@@ -0,0 +1,21 @@
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.DumbAwareAction
+import org.rust.lang.core.psi.impl.rustMod
+import org.rust.lang.core.psi.util.`super`
+
+class RustGoToSuperModAction : DumbAwareAction() {
+    override fun actionPerformed(e: AnActionEvent?) {
+        val project = e?.project ?: return
+        val psiFile = e?.getRequiredData(CommonDataKeys.PSI_FILE) ?: return
+        val superModFile = psiFile.rustMod?.`super`?.containingFile?.virtualFile ?: return
+
+        val descriptor = OpenFileDescriptor(project, superModFile, 0, 0)
+
+        FileEditorManager.getInstance(project).openTextEditor(descriptor, true) ?: return
```

Comment:
> maybe something i'm doing wrong.

Hm, I think it's me being wrong here :) I should have used `.navigate` in `UnresolvedModuleInspection`, and I shouldn't have used

```
val openFiles = FileEditorManager.getInstance(project).openFiles
```

In the tests. Thanks for pointing this out, I will fix it soon!

And about `GotoSuper`, I think we should stick to `navigate` here, and use `checkResultByFile`  (here is a [kotlinized version](https://github.com/intellij-rust/intellij-rust/blob/master/src/test/kotlin/org/rust/lang/RustTestCaseBase.kt#L21) and [example usage](https://github.com/intellij-rust/intellij-rust/blob/0b981b1670dbfedb5044d019922ead5c963209b3/src/test/kotlin/org/rust/lang/core/completion/RustCompletionTest.kt#L11-L16). Note how it is possible to pass the last lambda argument without parenthesis). I think it should be enough to test that `GotoSuper` works within one file, like this

```
//before.rs
mod foo {
    mod bar {
        fn foo() { <caret> }
    }
}

//after.rs
mod <caret>foo {
    mod bar {
        fn foo() {  }
    }
}
```